src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -2096,11 +2031,74 @@ static bool Set(out int x)
     }
 }
 ";
-            // Bug#529603: Won't Fix (Native no error)
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [WorkItem(529603, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/529603")]
+        [Theory]
+        [InlineData("true", "false")]
+        [InlineData("FIELD_TRUE", "FIELD_FALSE")]
+        [InlineData("LOCAL_TRUE", "LOCAL_FALSE")]
+        public void IfConditionalConstant(string @true, string @false)
+        {
+            var source = @"
+#pragma warning disable 219 // The variable is assigned but its value is never used
+#pragma warning disable 8321 // The local function is declared but never used
+
+class C
+{
+    const bool FIELD_TRUE = true;
+    const bool FIELD_FALSE = false;
+
+    static void M()
```

Comment:
I'd feel more comfortable using simple blocks to scope variables, rather than local functions (which introduce their own analysis quirks).

Reply:
Good idea. Done.